src/source/Ice/Network.c

```diff
@@ -340,36 +377,58 @@ STATUS getIpWithHostName(PCHAR hostname, PKvsIpAddress destIp)
     BOOL resolved = FALSE;
     struct sockaddr_in* ipv4Addr;
     struct sockaddr_in6* ipv6Addr;
+    struct in_addr inaddr;
+
+    CHAR addr[KVS_IP_ADDRESS_STRING_BUFFER_LEN + 1] = {'\0'};
+    CHAR address[KVS_IP_ADDRESS_STRING_BUFFER_LEN + 1] = {'\0'};
 
     CHK(hostname != NULL, STATUS_NULL_ARG);
 
-    errCode = getaddrinfo(hostname, NULL, NULL, &res);
-    if (errCode != 0) {
-        errStr = errCode == EAI_SYSTEM ? strerror(errno) : (PCHAR) gai_strerror(errCode);
-        CHK_ERR(FALSE, STATUS_RESOLVE_HOSTNAME_FAILED, "getaddrinfo() with errno %s", errStr);
+    // Adding this check in case we directly get an IP address. With the current usage pattern,
+    // there is no way this function would receive an address directly, but having this check
+    // in place anyways
+    if (isIpAddr(hostname)) {
+        MEMCPY(addr, hostname, STRLEN(hostname));
+    } else {
+        retStatus = getIpAddrFromDnsHostname(hostname, addr, STRLEN(hostname));
     }
 
-    for (rp = res; rp != NULL && !resolved; rp = rp->ai_next) {
-        if (rp->ai_family == AF_INET) {
-            ipv4Addr = (struct sockaddr_in*) rp->ai_addr;
-            destIp->family = KVS_IP_FAMILY_TYPE_IPV4;
-            MEMCPY(destIp->address, &ipv4Addr->sin_addr, IPV4_ADDRESS_LENGTH);
-            resolved = TRUE;
-        } else if (rp->ai_family == AF_INET6) {
-            ipv6Addr = (struct sockaddr_in6*) rp->ai_addr;
-            destIp->family = KVS_IP_FAMILY_TYPE_IPV6;
-            MEMCPY(destIp->address, &ipv6Addr->sin6_addr, IPV6_ADDRESS_LENGTH);
-            resolved = TRUE;
+    // Verify the generated address has the format x.x.x.x
+    if (!isIpAddr(addr) || retStatus != STATUS_SUCCESS) {
+        DLOGW("Parsing for address failed for %s, fallback to getaddrinfo", hostname);
+        errCode = getaddrinfo(hostname, NULL, NULL, &res);
+        if (errCode != 0) {
+            errStr = errCode == EAI_SYSTEM ? strerror(errno) : (PCHAR) gai_strerror(errCode);
+            CHK_ERR(FALSE, STATUS_RESOLVE_HOSTNAME_FAILED, "getaddrinfo() with errno %s", errStr);
```

Comment:
Validate no leaking of resources happen on these jumps to cleanup.

Reply:
We are good here. There arent any heap allocations and we free the `res` in the block it is used in.